### PR TITLE
Implement Code Coverage with Danger-Swift

### PR DESCRIPTION
**Pull request by AvdLee**
Migrates our code coverage report to Danger-Swift.

This is dependent on https://github.com/f-meloni/danger-swift-coverage/pull/21

### CODE REVIEW

File: Danger-Swift/Package.swift [hunk 1]
```diff
@@ -13,12 +13,15 @@ let package = Package(
     ],
     dependencies: [
 //        .package(url: "https://github.com/danger/swift", from: "2.0.7")
-        .package(path: "/Users/antoinevanderlee/Documents/GIT-Projects/Eigen/swift")
+        .package(path: "/Users/antoinevanderlee/Documents/GIT-Projects/Eigen/swift"),
```
Comment (kairadiagne) [inline Danger-Swift/Package.swift]: The package declaration contains the commented out packages and reference to local file paths. I assume for testing purposes. Was this intended to be committed?

File: Danger-Swift/Tests/WeTransferPRLinterTests/WeTransferLinterTests.swift [hunk 2]
```diff
@@ -2,24 +2,56 @@ import XCTest
...
+    private var buildFolder: Folder!
+
+    override func setUp() {
+        super.setUp()
+        buildFolder = try! Folder.current.createSubfolderIfNeeded(withName: ".filesTest")
+        try! buildFolder.empty()
+    }
+
     override func tearDown() {
+        try? buildFolder.delete()
```
Comment (kairadiagne) [inline Danger-Swift/Tests/WeTransferPRLinterTests/WeTransferLinterTests.swift]: Maybe we can force cast here just like I the `setup` method to make sure the build folder actually gets deleted. We could also use `XCTAssertNoThrow` to make sure the test do not crash.

File: Danger-Swift/Package.swift (hunk 1 above)
Comment (AvdLee) [reply Danger-Swift/Package.swift]: It was part of WIP as it was not yet released in the original repo. It is now, so I've updated it!

Comment (kairadiagne) [reply Danger-Swift/Package.swift]: Ah gotta, you mentioned it in the description.

File: Danger-Swift/Package.resolved [hunk 3]
```diff
@@ -1,6 +1,24 @@
 {
   "object": {
     "pins": [
+      {
+        "package": "DangerSwiftCoverage",
+        "repositoryURL": "https://github.com/f-meloni/danger-swift-coverage",
+        "state": {
+          "branch": null,
```
Comment (Boris-Em) [inline Danger-Swift/Package.resolved]: What does this default to? Master? 
Or does it work though the revision which is a commit id?

File: Danger-Swift/Sources/WeTransferPRLinter/CoverageReporter.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//
+
+import Foundation
+import DangerSwiftCoverage
+import Files
+
+public typealias XCResultBundle = Folder
+
+public protocol CoverageReporting {
+    static func reportCoverage(for xcresultBundle: XCResultBundle)
```
Comment (Boris-Em) [inline Danger-Swift/Sources/WeTransferPRLinter/CoverageReporter.swift]:
```suggestion
    static func reportCoverage(for xcResultBundle: XCResultBundle)
```

File: Danger-Swift/Package.resolved (hunk 3 above)
Comment (AvdLee) [reply Danger-Swift/Package.resolved]: It uses the tag!

**Comment by AvdLee** [comment]
@Boris-Em I'm going to merge this in. I assume it's alright for you, if not we can solve it in the base branch later!

**Pull request merged**